metrics/ci_integrations/test/integration/validation/model/validation_result_builder_test.dart

```diff
@@ -0,0 +1,155 @@
+      () {
+        expect(
+          () => ValidationResultBuilder.forFields(fields),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".setResult() throws an ArgumentError if the field is not available",
+      () {
+        final nonAvailableField = ConfigFieldStub('non available');
+
+        expect(
+          () => builder.setResult(
+            nonAvailableField,
+            FieldValidationResult.success(),
+          ),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".setResult() sets the given field validation result to the given field",
+      () {
+        builder.setResult(firstField, successResult);
+        builder.setResult(anotherField, failureResult);
+
+        final result = builder.build();
+        final validationResults = result.results;
+
+        expect(validationResults, containsPair(firstField, successResult));
+        expect(validationResults, containsPair(anotherField, failureResult));
+      },
+    );
+
+    test(
+      ".setEmptyResults() sets the empty field validation results to the given result",
+      () {
+        builder.setEmptyResults(successResult);
+
+        final result = builder.build();
+        final fieldValidationResults = result.results.values;
+
+        expect(fieldValidationResults, everyElement(equals(successResult)));
+      },
+    );
+
+    test(
+      ".setEmptyResults() does not override the non-empty field validation results",
+      () {
+        builder.setResult(firstField, failureResult);
+
+        builder.setEmptyResults(successResult);
+
+        final result = builder.build();
+        final validationResults = result.results;
+
+        expect(validationResults, containsPair(firstField, failureResult));
+      },
+    );
+
+    test(
+      ".build() throws a StateError if there are fields with not specified validation result",
+      () {
+        expect(
+          () => builder.build(),
+          throwsStateError,
+        );
+      },
+    );
+
+    test(
+      ".build() throws a StateError with a message that contains field names with not specified validation result",
+      () {
+        final emptyFields = [firstField, anotherField];
+
+        expect(
+          () => builder.build(),
+          throwsA(
+            predicate<StateError>(
+              (error) {
+                return emptyFields.every(
+                  (emptyField) => error.message.contains('$emptyField'),
+                );
+              },
+            ),
+          ),
+        );
+      },
+    );
+
+    test(
+      ".build() returns a validation result containing validation results that were set",
+      () {
+        builder.setResult(firstField, successResult);
+        builder.setResult(anotherField, failureResult);
+
+        final result = builder.build();
+        final validationResults = result.results;
+
+        expect(
+          validationResults,
+          equals(results),
+        );
```

Comment:
```suggestion
        expect(validationResults[firstField], equals(successResult));
        expect(validationResults[anotherField], equals(failureResult));
```
Would it be more reliable?